app/src/test/java/waiter/CommunicatorTest.java

```diff
@@ -0,0 +1,21 @@
+package waiter;
+
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class CommunicatorTest {
+
+    @Test
+    void establishesCommunication() throws IOException {
+        String userInput = "Doesn't matter whats here, simulating any user input message";
+        ListenerMock listenerMock = new ListenerMock(userInput);
+        MessengerMock messengerMock = new MessengerMock();
+        Communicator communicator = new Communicator();
+        communicator.communicate(listenerMock, messengerMock);
+        Connection calledWithConnection = messengerMock.calledWith;
```

Comment:
What is this `calledWith`?

Reply:
It is the clientConnectionMock that is set when messengerMock's transport method is called (the messengerMocks transport method is called with that clientConnectionMock), so that you an compare the user_input and the string that was set in that clientConnectionMock's write method

Reply:
I hope I explained this well 😄